### DISCUSSION TITLE
almost all data gone

### DISCUSSION THREAD

**Discussion by Axilotl17** [ppy/osu#17213] [Q&A] [answered]
all my beatmaps, collections, scores - all the beatmaps ive put hundreds of hours into, gone, overnight. i didn't do anything, i didn't even turn off my computer, or update osu, its just all gone upon closing and opening osu. i tried restarting, reinstalling, using an older version, and neither worked. I'm not sure where my beatmaps are stored, but if they're still on my computer id **REALLY** like them back. 

and yes, i did try restoring recently deleted beatmaps. 

[logs.zip](https://github.com/ppy/osu/files/8228747/logs.zip)
not sure if you'll find anything here as i did restart my computer. after i restarted, it gave me some screen about a database? and then threw some error i cant replicate again and i still dont have any beatmaps. 

![Screenshot from 2022-03-10 19-28-32](https://user-images.githubusercontent.com/78441708/157796617-51753e89-b08e-4433-88b9-f9e4c58d4976.png)
this was in my backups folder after the second time launching, after i restarted my computer. it wasn't there the first time.

**Comment by frenzibyte** [comment, marked as answer]
```
022-03-11 03:14:53 [verbose]: Realm compact failed with error Realms.Exceptions.RealmException: Provided schema version 13 is less than last set version 14.
2022-03-11 03:14:53 [verbose]: at Realms.NativeException.ThrowIfNecessary(Func`2 overrider)
2022-03-11 03:14:53 [verbose]: at Realms.SharedRealmHandle.Open(Configuration configuration, RealmSchema schema, Byte[] encryptionKey)
2022-03-11 03:14:53 [verbose]: at Realms.RealmConfiguration.CreateRealm()
2022-03-11 03:14:53 [verbose]: at Realms.Realm.GetInstance(RealmConfigurationBase config)
2022-03-11 03:14:53 [verbose]: at Realms.Realm.Compact(RealmConfigurationBase config)
2022-03-11 03:14:53 [verbose]: at osu.Game.Database.RealmAccess.BlockAllOperations() in /var/lib/buildkite-agent/builds/debian-gnu-linux-vm-2/ppy/osu/osu.Game/Database/RealmAccess.cs:line 617
2022-03-11 03:14:53 [verbose]: Creating full realm database backup at client_1646968493_corrupt.realm
```

You seem to have ran a different osu!lazer version and ended up with this error (this will be handled better next release, see https://github.com/ppy/osu/pull/17158), overwrite your current `client.realm` file with the backed up one (`client_1646968493_corrupt.realm`) and *ensure* you run the latest osu! version 2022.306.0.

If an issue still persists with running the database, do provide logs of that.

---

As an aside, has it been a while since you have now opened osu!(lazer)? `client.before_final_migration` looks to have been created at the same time you have run this game now, and that can only be the case when you have an existing `client.db` in case (which should not exist if you have run the game within the last few months).

**Reply by Axilotl17** [reply]
i opened osu lazer last night, and pretty much every day for the past month, so i dont know about been a while. also, where do i find the current client.realm?

**Reply by Axilotl17** [reply]
ive also got to go ill be back to try this in an hour about

**Reply by frenzibyte** [reply]
That's quite weird... the only other explaination coming to my mind to this is that you're suddenly running on a different osu!lazer folder. `client.realm` should exist wherever you have your osu!lazer folder located (probably in `~/.local/share/osu` as a default location I think?)

**Reply by Axilotl17** [reply]
i also cannot find `client_1646968493_corrupt.realm` in the first place, although i did find normal client.realm

![image](https://user-images.githubusercontent.com/78441708/157806023-aa137c4a-8ceb-436e-90bc-50e39c73ac5e.png)
only these

**Reply by peppy** [reply]
The `client.before_final_migration` should work fine. The timestamps are the same.

**Reply by Axilotl17** [reply]
i tried that, no beatmaps. i did find the right one i think although it doesn't work.

![image](https://user-images.githubusercontent.com/78441708/157825176-82290ca5-28dd-4881-966f-db3844dc3e42.png)

i get this error

[client_1646985039_corrupt.zip](https://github.com/ppy/osu/files/8230080/client_1646985039_corrupt.zip)
its this one im pretty sure, its the oldest corrupt one, and its the largest file size of all the clients, probably from all the beatmaps. if you can somehow salvage this, please, i would be extremely grateful. let me know if i can help.

**Reply by Axilotl17** [reply]
when i replace `client.realm` with the one i just sent i get above error

**Reply by Theighlin** [reply]
Are running the latest release (2022.306.0) after replacing `client.realm`? If you're not it will continue to corrupt it.
> overwrite your current client.realm file with the backed up one (client_1646968493_corrupt.realm) and ensure you run the latest osu! version 2022.306.0.

> If an issue still persists with running the database, do provide logs of that.

**Reply by frenzibyte** [reply]
@Axilotl17 Your database is working just fine on my end, so please do as mentioned above. Confirm you're running 2022.306.0 and attach logs of the error you have got.

**Reply by Axilotl17** [reply]
Okay I will try; the aur didn’t have 306 last I checked so I was using the app image and I’m unsure which one i was launching yesterday

**Reply by Axilotl17** [reply]
it worked, thank you so so much!

**Reply by frenzibyte** [reply]
Keep in mind the AUR package is not official (hence not mentioned in the README), osu!(lazer) will eventually support updaters in linux (see https://github.com/ppy/osu/discussions/16908#discussioncomment-2336391)
